Sources/Testing/Events/Event.swift

```diff
@@ -447,3 +448,51 @@ extension Event.Kind {
     }
   }
 }
+
+extension Event.Context {
+
+  /// A serializable type which provides context about a posted ``Event``.
+  ///
+  @_spi(ForToolsIntegrationOnly)
+  public struct Snapshot: Sendable, Codable {
+    /// A snapshot of the test for which this instance's associated ``Event``
+    /// occurred, if any.
+    ///
+    /// If an event occurred independently of any test, or if the running test
+    /// cannot be determined, the value of this property is `nil`.
+    public var test: Test.Snapshot?
+
+    /// A snapshot of the test case for which this instance's associated
+    /// ``Event`` occurred, if any.
+    ///
+    /// The test case indicates which element in the iterated sequence is
+    /// associated with this event. For non-parameterized tests, a single test
+    /// case is synthesized. For test suite types (as opposed to test
+    /// functions), the value of this property is `nil`.
+    public var testCase: Test.Case.Snapshot?
+
+    /// Initialize a new instance of this type.
+    ///
+    /// - Parameters:
+    ///   - test: The test for which this instance's associated event occurred,
+    ///     if any.
+    ///   - testCase: The test case for which this instance's associated event
+    ///     occurred, if any.
+    public init(
```

Comment:
Why do we need this in addition to `public init(snapshotting context: borrowing Context)`? Do we expect somebody would ever need to initialize an instance of the snapshot type without an instance of the type being snapshotted?

Reply:
Yes.

Reply:
Why? When?